tests/dates/test_dates.py

```diff
@@ -37,6 +45,49 @@ def test_get_numerical_date_from_value_ambiguous_day(self):
             == pytest.approx(2000.083, abs=1e-3)
         )
 
+    @pytest.mark.parametrize(
+        "value",
+        [
+            "2000-001-XX",
+            "2000-001",
+        ],
+    )
+    def test_get_numerical_date_from_value_ambiguous_day_error(self, value):
+        with pytest.raises(dates.errors.InvalidDate, match="Invalid date '2000-001-XX': Date does not match format `%Y-%m-%d`"):
+            dates.get_numerical_date_from_value(value, "%Y-%m-%d")
+
+    @pytest.mark.parametrize(
+        "value",
+        [
+            "2000-01-01",
+            "2000-1-1",
+            "2000-01-1",
+            "2000-1-01",
+            "2000-001-01",
+            "2000-001-001",
+        ],
+    )
+    def test_get_numerical_date_from_value_iso_like_date(self, value):
+        assert (
+            dates.get_numerical_date_from_value(value, "%Y-%m-%d")
+            == pytest.approx(dates.numeric_date(datetime.date(year=2000, month=1, day=1)), abs=1e-3)
+            == pytest.approx(2000.001, abs=1e-3)
+        )
+
+    @pytest.mark.parametrize(
+        "value",
+        [
+            "2000-00-01",
+            "2000-13-01",
+            "2000-100-01",
+            "2000-01-00",
+            "2000-01-32",
+            "2000-01-100",
+        ],
+    )
+    def test_get_numerical_date_silent_error(self, value):
+        assert dates.get_numerical_date_from_value(value, "%Y-%m-%d") == None
```

Comment:
_non-blocking_

Hmm, there's nothing in ncov-ingest's [format_dates](https://github.com/nextstrain/ncov-ingest/blob/fc91ecca68c3a96239a5263b803bf1565da0e025/lib/utils/transform.py#L58) guarding against bad dates like these, so does that mean the error from `augur filter` reported in https://github.com/nextstrain/augur/issues/1785 could potentially happen again? 

I'm seeing more reason to update ncov-ingest...

Reply:
I ran these changes through the latest full GISAID metadata and it completed successfully. I think it's unlikely that we'll see any of these weird dates, but it's not impossible.

Reply:
Since I'm already planning to release this as a major version, I thought we might as well fail on error in this case instead of silent error which then could fail with a more cryptic error such as #1785.

Relevant changes:

https://github.com/nextstrain/augur/blob/047da3f6a59965fc602e564e6a696356aeeae25e/CHANGES.md#L10

https://github.com/nextstrain/augur/blob/047da3f6a59965fc602e564e6a696356aeeae25e/tests/dates/test_dates.py#L80-L85